core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/DiagramServices.java

```diff
@@ -2100,13 +2102,20 @@ public void refreshRepresentationOfEditor(IEditorPart editor) {
    * @return
    */
   public EditPart getEditPart(DDiagramElement diagramElement) {
-    IEditorPart editor = EclipseUIUtil.getActiveEditor();
-    if (editor instanceof DiagramEditor) {
-      Session session = new EObjectQuery(diagramElement).getSession();
+    DDiagram parentDiagram = diagramElement == null ? null : diagramElement.getParentDiagram();
+    Session session = parentDiagram == null ? null : new EObjectQuery(parentDiagram).getSession();
```

Comment:
If parentDiagram is null, we can return null immediately instead of doing additional computations

Reply:
Each computation has a guard and having multiple `return null` in this method will reduce readability compared to the current state.

Do not hesitate to make this step on your side on top of this PR as if you which to go on this direction this would have been true on the previous code state

Reply:
Please consider the "return early" pattern.

Reply:
It is not a return early but now it is more readable. Looks good to go.